lib/tools.js

```diff
@@ -2207,9 +2207,9 @@ function getInstances(adapter, objects, withObjects, callback) {
 /**
  * Checks if the given callback is a function and if so calls it with the given parameter immediately, else a resolved Promise is returned
  *
- * @param {function(...any=):void|undefined} callback - callback function to be executed
- * @param {...any} [args] - as many arguments as needed, which will be returned by the callback function or by the Promise
- * @returns {Promise} - if Promise is resolved with multiple arguments, an array is returned
+ * @param {(...args: any[]) => void | null | undefined} callback - callback function to be executed
+ * @param {any[]} args - as many arguments as needed, which will be returned by the callback function or by the Promise
+ * @returns {Promise<any>} - if Promise is resolved with multiple arguments, an array is returned
```

Comment:
Correct would be `Promise<any> | void` but this is awkward since TypeScript does not correctly understand overload signatures in JSDoc. So I'd leave it like this.